shared/router-v2/router.native.tsx

```diff
@@ -278,106 +278,236 @@ const useBarStyle = () => {
+            },
+          }
+          setInitialState({
+            index: 0,
+            routes: [
+              {
+                name: 'loggedIn',
+                state: {
+                  index: 0,
+                  routeNames: [Tabs.peopleTab],
+                  routes: [{name: Tabs.peopleTab, ...initialTabState}],
+                },
+              },
+            ],
+          })
+        }
+      } else if (startupTab || startupConversation) {
+        try {
+          const tab = startupConversation ? Tabs.chatTab : startupTab
+          C.Chat.useState_.getState().dispatch.unboxRows([startupConversation])
+          C.Chat.getConvoState_(startupConversation).dispatch.loadMoreMessages({
+            reason: 'savedLastState',
+          })
+
+          const initialTabState = startupConversation
+            ? {
                 state: {
                   index: 1,
-                  routes: [{name: 'chatRoot'}, {name: 'chatConversation', params: {conversationIDKey}}],
+                  routes: [
+                    {name: 'chatRoot'},
+                    {name: 'chatConversation', params: {conversationIDKey: startupConversation}},
+                  ],
                 },
               }
-            })
-            setInitialState(next)
-          } catch {}
-          setShowNav(true)
-        } else {
-          setTimeout(() => {
-            C.useDeepLinksState.getState().dispatch.handleAppLink(url)
-            setTimeout(() => {
-              setShowNav(true)
-            }, 500)
-          }, 1)
-        }
-      } else {
-        setShowNav(true)
+            : {}
+
+          setInitialState({
+            index: 0,
+            routes: [
+              {
+                name: 'loggedIn',
+                state: {
+                  index: 0,
+                  routeNames: [tab],
+                  routes: [{name: tab, ...initialTabState}],
+                },
+              },
+            ],
+          })
+        } catch {}
       }
     }
+
+    const f = async () => {
+      await loadInitialURL()
+      setInitialStateState('loaded')
+    }
+
     C.ignorePromise(f())
-  }, [initialNav, ready, setAppState])
-  return {initialState, onReady, setShowNav, showNav}
+  }, [
+    loggedIn,
+    startupLoaded,
+    initialState,
+    initialStateState,
+    androidShare,
+    showMonster,
+    startupConversation,
+    startupFollowUser,
+    startupTab,
+  ])
+
+  return {initialState, initialStateState}
+}
+
+// on android we rerender everything on dark mode changes
+const useRootKey = () => {
+  const [rootKey, setRootKey] = React.useState('')
+  const isDarkMode = Kb.Styles.useIsDarkMode()
+  React.useEffect(() => {
+    if (!C.isAndroid) return
+    setRootKey(isDarkMode ? 'android-dark' : 'android-light')
+  }, [isDarkMode])
+
+  return rootKey
 }
 
 const RNApp = React.memo(function RNApp() {
-  const s = Shared.useShared()
```

Comment:
this is where we do a lot of simplifying. on desktop+ios we use a native darkmode solution. on android we use a key to just rerender everything
previously we needed the nav rendered, then loaded, then tried to reload it intelligently
now we load and then render nav after all the initial state is known once